yt_dlp/extractor/niconico.py

```diff
@@ -693,7 +693,7 @@ def _real_extract(self, url):
             title = unescapeHTML(title)
         playlist = [
             self.url_result(f'https://www.nicovideo.jp/watch/{v_id}', video_id=v_id)
-            for v_id in re.findall(r'data-href=[\'"](?:https://www\.nicovideo\.jp)?/watch/([a-z0-9]+)', webpage)]
+            for v_id in re.findall(r'[\'"]url[\'"]:[\'"]https:\\/\\/www.nicovideo.jp\\/watch\\/([a-z0-9]+)', webpage)]
```

Comment:
Maybe it should parse JSON in `<script type="application/ld+json">` instead?

Reply:
Tried it and it seems to be cleaner yeah. Any issues with the new code?

Reply:
I don't think it has any issue